runtime/common/src/signature.rs

```diff
@@ -0,0 +1,53 @@
+use common_primitives::utils::wrap_binary_data;
+use sp_runtime::{traits::Verify, AccountId32, MultiSignature};
+use sp_std::vec::Vec;
+
+pub fn check_signature(signature: &MultiSignature, signer: AccountId32, payload: Vec<u8>) -> bool {
+	let verify_signature = |payload: &[u8]| signature.verify(payload, &signer.clone().into());
+
+	if verify_signature(&payload) {
+		return true;
+	}
+
+	let wrapped_payload = wrap_binary_data(payload);
+	verify_signature(&wrapped_payload)
```

Comment:
nit: I would prefer if we check the wrapped one first as a minor optimization due to most of our signatures are being wrapped.

Reply:
> nit: I would prefer if we check the wrapped one first as a minor optimization due to most of our signatures are being wrapped.

https://github.com/frequency-chain/frequency/pull/2169#discussion_r1784764586